net/unbound/files/unbound.sh

```diff
@@ -1067,28 +1065,33 @@ unbound_start() {
 
     unbound_control
   fi
+
+  if unbound_ismain ; then
+    # set resolver file to local.
+    rm -f /tmp/resolv.conf
+
+    {
+      echo "nameserver 127.0.0.1"
+      echo "nameserver ::1"
+      echo "search $UNBOUND_TXT_DOMAIN"
+    } > /tmp/resolv.conf
+  fi
 }
 
 ##############################################################################
 
 unbound_stop() {
-  local resolvsym=0
-
-  rootzone_update
-
-
-  if [ ! -x /usr/sbin/dnsmasq -o ! -x /etc/init.d/dnsmasq ] ; then
-    resolvsym=1
-  else
-    /etc/init.d/dnsmasq enabled || resolvsym=1
-  fi
+  config_load unbound
+  config_foreach unbound_uci unbound
```

Comment:
Indeed it's a bit overdone to read-in all the UCI just for the port number.
Service stops normally aren't performance sensitive so performance weighted less than clean code (that adheres to how UCI is handled in the rest of the file.

Racing conditions, please explain, I lost you here ... Can the code path for UCI-change reload hit unbound_stop() ?
(for myself I wrote a little script to switch the dhcp linkage and with the scheme stop-reconf-start I do not really see how that could race against ...).

At my age cryptic riddles ("that is case $( cat ) in ))) esac") get a bit harder to grasp, so please, what do you mean ? Once I myself do understand why and how the code can be improved, I'll happily fix that or leave a comment to remember.

OT: is 5353 not kind of reserved for mdns ?